### PR TITLE
chore: sync with upstream `fef2098`

### PR DESCRIPTION
**Pull request by Karrq**
Sync with upstream's https://github.com/foundry-rs/foundry/commit/fef2098

Related PRs:
* #768 
* #754
* #736
* #770

### CODE REVIEW

**Review by Jrigada — APPROVED** [review]

**Review by tomimor — APPROVED** [review]

**Review by elfedy — APPROVED** [review]

File: crates/cast/bin/cmd/send.rs [hunk 1]
```diff
@@ -197,20 +197,23 @@ impl SendTxArgs {
...
 
                 tx::validate_from_address(eth.wallet.from, from)?;
 
                 // Zksync transaction
-                let (tx, _) = builder.build(&zk_signer).await?;
+                let (tx, _) = builder.build(&signer).await?;
+                let signer = Arc::new(signer);
 
-                let zk_wallet = ZksyncWallet::from(zk_signer);
+                let zk_wallet = ZksyncWallet::from(signer.clone());
```
Comment (nbaztec) [inline crates/cast/bin/cmd/send.rs:209]: Didn't you say this panics in ledger?

**Review by nbaztec — APPROVED** [review]

File: crates/cast/bin/cmd/send.rs (hunk 1 above)
Comment (Karrq) [reply crates/cast/bin/cmd/send.rs:209]: The initialization is, not sure about a clone, but I put it behind an Arc to ensure we don’t actually clone the data

**Comment by Karrq** [comment]
Status: testing with ZK stack

**Comment by Jrigada** [comment]
Hey! I just tested it with zkstack through the ecosystem initialization and the chain also to test the L2 and L1 deployment scripts. Everything green on my end. Any more tests that would be beneficial to run @dutterbutter ? Great work Francesco!

**Comment by Karrq** [comment]
Waiting to merge this on monday

**Pull request merged**